DotNet.Extensions.Require.Test/ChecksTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace DotNet.Extensions.Require.Test;
+
+public sealed class ChecksTests
+{
+    [Test]
+    public void Check_ThrowsCheckFailed_WhenRequirementIsNotSatisfied() =>
+        Assert.Throws<CheckFailed>(
+            () => 1.Check(v => v != 1, expectation: "message")
+        )?.WithMessage("message");
+
+    [Test]
+    public void Check_DoesNotThrow_WhenRequirementIsSatisfied() =>
+        Assert.DoesNotThrow(
+            () => 1.Check(v => v == 1, expectation: string.Empty)
+        );
+
+    [Test]
+    public void Check_ThrowsBuiltException_WhenRequirementIsNotSatisfied() =>
+        Assert.Throws<SomeExceptionType>(
+            () => 1.Check(v => v != 1, () => new SomeExceptionType("message"))
+        )?.WithMessage("message");
+
+    [Test]
+    public void Check_CanConstructException_WithTheValue() =>
+        Assert.Throws<SomeExceptionType>(
+            () => 1.Check(v => v != 1, v => new SomeExceptionType($"message with {v}"))
+        )?.WithMessage("message with 1");
+
+    [Test]
+    public void Check_DoesNotBuildException_WhenRequirementIsSatisfied()
+    {
+        Assert.DoesNotThrow(
+            () => 1.Check(v => v == 1, expectation: () => throw new NUnitException("must not be called"))
```

Comment:
Warum nimmt er hier die Überladung mit dem `Func<string>`-Parameter? Ist das richtig?

Reply:
yes, I want to check that neither the `CheckFailed` nor the custom exception type is build/constructed when the requirement is satisfied. I put it in the same test as I couldn't find a good name the distinguishes both. In the end, it's the same behavior.